bundles/com.amazonaws.eclipse.core/src/com/amazonaws/eclipse/core/mobileanalytics/ToolkitEvent.java

```diff
@@ -14,141 +14,144 @@
-     */
-    public boolean isValid() {
-        if (session == null) {
-            return false;
-        }
-        if (session.getId() == null) {
-            return false;
-        }
-        if (session.getStartTimestamp() == null) {
-            return false;
-        }
-        if (eventType == null || eventType.isEmpty()) {
-            return false;
-        }
-        if (timestamp == null) {
-            return false;
-        }
-        if (attributes.size() + metrics.size() > Constants.MAX_ATTRIBUTES_AND_METRICS_PER_EVENT) {
-            return false;
-        }
-        for (Entry<String, String> attribute : attributes.entrySet()) {
-            if (attribute.getKey().length() > Constants.MAX_ATTRIBUTE_OR_METRIC_NAME_LENGTH) {
-                return false;
-            }
-            if (attribute.getValue() == null) {
-                return false;
-            }
-            if (attribute.getValue().length() > Constants.MAX_ATTRIBUTE_VALUE_LENGTH) {
-                return false;
-            }
-        }
-        for (Entry<String, Double> metric : metrics.entrySet()) {
-            if (metric.getKey().length() > Constants.MAX_ATTRIBUTE_OR_METRIC_NAME_LENGTH) {
-                return false;
-            }
-            if (metric.getValue() == null) {
-                return false;
-            }
-        }
-        return true;
-    }
-
-    /**
-     * The constructor is intentionally marked as private; caller should use
-     * ToolkitEventBuilder to create event instance
-     */
-    private ToolkitEvent() {}
-
-    public static class ToolkitEventBuilder {
-
-        private final ToolkitEvent event = new ToolkitEvent();
-
-        public ToolkitEventBuilder(ToolkitSession session) {
-            this.event.session = session;
-        }
-
-        public ToolkitEventBuilder setEventType(String eventType) {
-            this.event.eventType = eventType;
-            return this;
-        }
-
-        /**
-         * If not specified, the timestamp is by default set to the current
-         * time.
-         */
-        public ToolkitEventBuilder setTimestamp(Date timestamp) {
-            this.event.timestamp = timestamp;
-            return this;
-        }
-
-        public ToolkitEventBuilder addAttribute(String name, String value) {
-            this.event.attributes.put(name, value);
-            return this;
-        }
-
-        public ToolkitEventBuilder addMetric(String name, double value) {
-            this.event.metrics.put(name, value);
-            return this;
-        }
-
-        public ToolkitEventBuilder addBooleanMetric(String name, boolean value) {
-            this.event.metrics.put(name, value ? 1.0 : 0.0);
-            return this;
-        }
-
-        public ToolkitEvent build() {
-            if (this.event.timestamp == null) {
-                this.event.timestamp = new Date();
-            }
-            return this.event;
-        }
-
-    }
-
+	private ToolkitSession session;
```

Comment:
looks like some files were switched to tabs. But most files are indented with spaces?

Reply:
Didn't realize the default was tabs, that reduced the diff size quite a bit